utils.js

```diff
@@ -391,6 +391,29 @@ function isDomElement(thing) {
     return thing.nodeName !== undefined;
 }
 
+/**
+ * Checks whether any of the element's attributes satisfy some function.
+ * Default is to check all attributes, unless specific attributes are specified.
+ * If an attribute has an array of values (e.g. class attribute), will apply
+ * the searchFunction to each one.
+ * @arg fnode {object} fnode whose attributes you want to search
+ * @arg searchFunction {function} function to apply to each attribute
+ * @arg attrs {string} the attributes you want to search. If none provided, default is to search all.
+ * @return true if any of the attributes satisfy some condition
+ */
+function searchAttributes(fnode, searchFunction, ...attrs) {
+    // if attrs argument not provided, default is to search all attributes
+    const attributes = attrs.length === 0? Array.from(fnode.element.attributes).map(a => a.name) : attrs;
+    for (let i = 0; i < attributes.length; i++) {
+        const attr = fnode.element.getAttribute(attributes[i]);
+        // If the attribute is an array, apply the scoring function to each element
+        if (attr && ((attr.isArray && attr.some(searchFunction)) || searchFunction(attr))) {
```

Comment:
In what case would `attr` be falsey? Could that be something that searchers may want to be able to search for?

Reply:
When a user specifies some attribute they want to check, but the node doesn't actually have that attribute, attr would be falsey. I think searchers could check this easily just using element.hasAttribute before they put it into this function, I just wrote this as a precaution.

Reply:
I think it would be valuable to have a test that doesn't find anything.

Is it possible for the search to return multiple things? That would also be a useful test if so.